vignettes/adcoeq.Rmd

```diff
@@ -0,0 +1,215 @@
+qs_metabolic <- admiralmetabolic::qs_metabolic
+admiral_adsl <- admiral::admiral_adsl
+
+dm <- convert_blanks_to_na(dm_metabolic)
+qs <- convert_blanks_to_na(qs_metabolic)
+admiral_adsl <- convert_blanks_to_na(admiral_adsl)
+```
+
+Within this vignette, `DM` is used as the basis for `ADSL`:
+```{r eval=TRUE}
+# Retrieve required variables from admiral ADSL for this vignette that are not present in DM dataset
+adsl <- dm %>%
+  select(-DOMAIN) %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+## Original Items {#original_items}
+
+The original items, i.e. the answers to the questionnaire questions, can be handled in the same way as in an [{admiral} BDS finding ADaM](https://pharmaverse.github.io/admiral/articles/bds_finding.html). 
+
+```{r eval=TRUE}
+adcoeq1 <- qs %>%
+  # Add ADSL variables
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, TRTEDT, TRT01P, TRT01A),
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # Add analysis parameter variables
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  ) %>%
+  # Add timing variables
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  mutate(
+    AVISIT = if_else(ADT <= TRTSDT, "BASELINE", VISIT),
+    AVISITN = if_else(ADT <= TRTSDT, 0, VISITNUM)
+  ) 
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq1, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT)
+)
+```
+
+If the numeric values of the answers in `QSSTRESN` values are not ready to be used for deriving
+scores and require transformation, it is recommended that `QSSTRESN` is kept in
+the ADaM dataset for traceability, and the transformed value is stored in
+`AVAL`, since that's what will be used for the score calculation.
+For the CoEQ item 6 has to be handled in such way as in the "Positive Mood" section high values are supposed to indicate a positive outcome (original direction of item 6: 0 = Not at all anxious = Positive, 100 = Extremely anxious = Negative):
+
+```{r eval=TRUE}
+adcoeq2 <- adcoeq1 %>%
+  # Add analysis value variables
+  mutate(
+    AVAL = if_else(PARAMCD == "COEQ06", 100-QSSTRESN, QSSTRESN),
+    AVALC = if_else(PARAMCD == "COEQ20", QSORRES, NA_character_)
+  ) 
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq2, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT, AVALC, AVAL),
+  filter = PARAMCD %in% c("COEQ01", "COEQ02", "COEQ03", "COEQ04", "COEQ05", "COEQ06", "COEQ07", "COEQ08", "COEQ09", "COEQ20")
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
```

Comment:
```suggestion
For this example, we handle unscheduled visits as normal visits, and hence, don't need to perform any derivations. For deriving visits based on
time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
```

I am somewhat unsure why we write how we handle unscheduled visits, when we don't actually do anything. I have tried to formulate something, but unsure if it makes sense?

Reply:
I would be in favour of cutting entirely

Reply:
I would suggest we keep the general reference ("For deriving visits based on time-windows, see ..."), but we remove the first sentence ("We handle unscheduled visits as normal visits."). What do you think?

Reply:
Agree

Reply:
Done